src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.cs

```diff
@@ -69,9 +69,12 @@ internal sealed partial class SolutionCompilationState
         FrozenSourceGeneratedDocumentStates = frozenSourceGeneratedDocumentStates;
 
         // when solution state is changed, we recalculate its checksum
-        _lazyChecksums = AsyncLazy.Create(static (self, c) =>
-            self.ComputeChecksumsAsync(projectId: null, c),
-            arg: this);
+        _lazyChecksums = AsyncLazy.Create(static async (self, cancellationToken) =>
```

Comment:
style preference.  i like to use `cancellationToken` for callbacks.  that way if there is a cancellationToken in scope, it will hide it and we don't accidentally capture the wrong thing.  less relevant for static lambdas.  but i like to be consistent with that.